gc/gc-base/src/main/java/org/projectnessie/gc/base/DistributedIdentifyContents.java

```diff
@@ -93,29 +91,31 @@ private static int getPartitionsCount(GCParams gcParams, List<Reference> referen
 
   private static Map<String, ContentBloomFilter> mergeLiveContentResults(
       List<Map<String, ContentBloomFilter>> bloomFilterMaps, double bloomFilterFpp) {
-    Set<String> mergedContentIds = new HashSet<>();
     Map<String, ContentBloomFilter> output = new HashMap<>();
     bloomFilterMaps.forEach(
         map ->
             map.forEach(
                 (k, v) -> {
                   if (output.containsKey(k)) {
                     output.get(k).merge(v);
-                    mergedContentIds.add(k);
                   } else {
                     output.put(k, v);
                   }
                 }));
     // Since we merged bloom filters log in case their quality deteriorated
-    mergedContentIds.stream()
+    output.entrySet().stream()
         .forEach(
-            contentId -> {
-              double fpp = output.get(contentId).getExpectedFpp();
-              if (fpp > bloomFilterFpp) {
-                LOGGER.info(
-                    "Fpp of ContentBloomFilter for '{}': {}",
-                    contentId,
-                    String.format("%.3f", fpp));
+            e -> {
+              ContentBloomFilter bloomFilter = e.getValue();
+              if (bloomFilter.wasMerged()) {
```

Comment:
Nit: can use `Stream.filter()` (also for the `if` below)

Reply:
used for the first one but since we only want to call `expectedFpp()` once, mapping the entries to new `bstractMap.SimpleImmutableEntry` or `Pair` just to be able to use filter for the 2nd if, does not improve readability imo.